### ISSUE TITLE
lazySingleton module is not registered as lazySingleton but registered as factory.

### ISSUE THREAD

**Issue by AliHoseinpoor** [Milad-Akarie/injectable#176]
I want to register the Dio package as lazySingleton in the app but when I write the code it is registered in the injection.config file as factory.
here my code :
```
import 'injection.config.dart';

final GetIt getIt = GetIt.instance;

@injectableInit
Future<void> configureInjection(String env) async {
  await $initGetIt(getIt, environment: env);
}

@module
abstract class InjectableModule {
 
  @lazySingleton
  Dio get dio => Dio();

  @preResolve
  Future<SharedPreferences> get prefs => SharedPreferences.getInstance();
}

```
and here my injection.config file : 
```
import 'package:dio/dio.dart' as _i3;
import 'package:get_it/get_it.dart' as _i1;
import 'package:injectable/injectable.dart' as _i2;
import 'package:shared_preferences/shared_preferences.dart' as _i4;

import 'injection.dart' as _i5; // ignore_for_file: unnecessary_lambdas

// ignore_for_file: lines_longer_than_80_chars
/// initializes the registration of provided dependencies inside of [GetIt]
Future<_i1.GetIt> $initGetIt(_i1.GetIt get,
    {String environment, _i2.EnvironmentFilter environmentFilter}) async {
  final gh = _i2.GetItHelper(get, environment, environmentFilter);
  final injectableModule = _$InjectableModule();
  gh.factory<_i3.Dio>(() => injectableModule.dio);
  await gh.factoryAsync<_i4.SharedPreferences>(() => injectableModule.prefs,
      preResolve: true);
  return get;
}

class _$InjectableModule extends _i5.InjectableModule {}

```

It is now clear in the code that Dio is registered with the factory not as lazySingleton.

**Comment by Milad-Akarie** [comment]
@AliHoseinpoor   fixed in `injectable_generator: ^1.1.1` please confirm and close. 
Thanks for reporting.

**Comment by AliHoseinpoor** [comment]
> @AliHoseinpoor fixed in `injectable_generator: ^1.1.1` please confirm and close.
> Thanks for reporting.

Yes, this problem has been fixed in version 1.1.1. 💪💪
Thank's You .
